analog-in/cpp/analog-in.cpp

```diff
@@ -35,8 +35,23 @@
 using namespace std;
 using namespace mraa;
 
+// check if running as root
+void CheckRoot(void)
+{
+	int euid = geteuid();
+	if (euid) {
+		cerr << "This project uses Mraa I/O operations, but you're not running as 'root'.\n"
+				"The IO operations below might fail.\n"
+				"See the project's Readme for more info.\n\n";
```

Comment:
It is not part of this JIRA but it might be convenient for you to implement it alongside with 
https://jira01.devtools.intel.com/browse/ISS-1796
Your call.

Reply:
As far as I understood I need to add some delay into rootCheck func and "unknownPlatformMessage" before return (2-3 sec ?)